types/node/events.d.ts

```diff
@@ -856,7 +913,14 @@ declare module "events" {
                  * ```
                  * @since v0.1.26
                  */
-                emit<K>(eventName: Key<K, T>, ...args: Args<K, T>): boolean;
+                emit<EventName extends EventNames<Events>>(
+                    eventName: EventName,
+                    ...args: Args<Events, EventName>
+                ): boolean;
+                emit<EventName extends string | symbol>(
+                    eventName: EventName,
+                    ...args: Args<Events, EventName>
+                ): boolean;
```

Comment:
Can this be changed from the doubled signature to a single one using `EventNameParam`?

Reply:
Only if numbers / bigints are truly unsupported. They are allowed as object keys and event emitters will technically allow anything that can be accessed by index. Combining these makes any number / bigint keys emit args take `...any[]` instead of what their eventmap describes.